src/components/plots/plot.ts

```diff
@@ -268,24 +269,40 @@ module Plottable {
      * Updates the extents associated with each attribute, then autodomains all scales the Plot uses.
      */
     protected _updateExtents() {
-      this._attrBindings.forEach((attr) => this._updateExtentsForKey(attr, this._attrBindings, this._attrExtents));
-      this._propertyExtents.forEach((property) => this._updateExtentsForKey(property, this._propertyBindings, this._propertyExtents));
+      this._attrBindings.forEach((attr) => this._updateExtentsForAttr(attr));
+      this._propertyExtents.forEach((property) => this._updateExtentsForProperty(property));
       this._scales().forEach((scale) => scale._autoDomainIfAutomaticMode());
     }
 
-    private _updateExtentsForKey(key: string, bindings: D3.Map<Plots.AccessorScaleBinding<any, any>>, extents: D3.Map<any[]>) {
+    private _updateExtentsForAttr(attr: string) {
+      this._updateExtentsForKey(attr, this._attrBindings, this._attrExtents);
+    }
+
+    protected _updateExtentsForProperty(property: string) {
+      this._updateExtentsForKey(property, this._propertyBindings, this._propertyExtents, this._filterForProperty(property));
+    }
+
+    protected _filterForProperty(property: string): _Accessor {
+      return null;
+    }
+
+    private _updateExtentsForKey(key: string, bindings: D3.Map<Plots.AccessorScaleBinding<any, any>>,
+        extents: D3.Map<any[]>, filter?: _Accessor) {
       var accScaleBinding = bindings.get(key);
       if (accScaleBinding.accessor == null) { return; }
       extents.set(key, this._datasetKeysInOrder.map((key) => {
         var plotDatasetKey = this._key2PlotDatasetKey.get(key);
         var dataset = plotDatasetKey.dataset;
         var plotMetadata = plotDatasetKey.plotMetadata;
-        return this._computeExtent(dataset, accScaleBinding.accessor, plotMetadata);
+        return this._computeExtent(dataset, accScaleBinding.accessor, plotMetadata, filter);
       }));
```

Comment:
The above code looks fine but it is a bit messy... However I can't think of much to clean it up...

That said, I feel that having an optional parameter in filter is messy since filter could now be `null`, `undefined` or valid

Should `_updateExtentsForAttr()` for now pass in null and then have `_updateExtentsForKey()` have the parameter be required?

Alternative clean solutions are preferred but I'm unsure